tls/s2n_record_read.c

```diff
@@ -106,12 +106,20 @@ int s2n_record_header_parse(
     S2N_ERROR_IF(conn->actual_protocol_version_established && MIN(conn->actual_protocol_version, S2N_TLS12) /* check against legacy record version (1.2) in tls 1.3 */
                             != version,
             S2N_ERR_BAD_MESSAGE);
-    POSIX_GUARD(s2n_stuffer_read_uint16(in, fragment_length));
 
-    /* Some servers send fragments that are above the maximum length.  (e.g.
+    /*
+     *= https://tools.ietf.org/rfc/rfc8446#section-5.1
+     *= type=exception
+     *= reason=Incorrect implementations exist in the wild. Ignoring instead.
+     *# The length MUST NOT exceed 2^14 bytes.  An
+     *# endpoint that receives a record that exceeds this length MUST
+     *# terminate the connection with a "record_overflow" alert.
+     *
+     * Some servers send fragments that are above the maximum length.  (e.g.
      * Openssl 1.0.1, so we don't check if the fragment length is >
      * S2N_TLS_MAXIMUM_FRAGMENT_LENGTH. The on-the-wire max is 65k
```

Comment:
nit
```suggestion
     * Some servers send fragments that are above the maximum length (e.g.
     * Openssl 1.0.1). So we don't check if the fragment length is >
     * S2N_TLS_MAXIMUM_FRAGMENT_LENGTH. The on-the-wire max is 65k.
```

Reply:
Also I'm not sure I understand what the on-the-wire max means in this comment. Isn't the 65k maximum enforced by the uint16 record length field? Should the comment say this instead?

Reply:
I believe `on-the-wire max` is talking about the theoretical max for TCP/IP 65K but also TLS which parses record len as a u16.

Reply:
TCP/IP have nothing to do with TLS record sizes. TCP just treats it all as an ordered bytestream.

Reply:
I was suggesting that possibly “on-the-wire” meant a single payload delivered by the TCP/IP stack. Since TCP/IP can have a max up to 2^16, that would be the actual max upper limit of a TLS record size.

Do you have a better interpretation for the “The on-the-wire max is 65k” comment from before?

Reply:
But the size of a single TCP payload has no impact on the size of a TLS record, so that can't be the meaning of "on-the-wire max".